flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/transform/PreTransformOperator.java

```diff
@@ -270,6 +273,24 @@ public void close() throws Exception {
     @Override
     public void processElement(StreamRecord<Event> element) throws Exception {
         Event event = element.getValue();
+        if (event instanceof ChangeEvent) {
```

Comment:
If I understand correctly, after this change, MySQL source will always emit `CreateTableEvent`  based on current tables' schema, before sending any `DataChangeEvent`s.

As `event instanceof ChangeEvent` (L294) is `true`, we will emit a `CreateTableEvent` restored from `PreTransformOp`'s state.

But the following `event instanceof CreateTableEvent` is also `true`. Are we accidentally emitting duplicate and inconsistent `CreateTableEvent`s to the downstream?

Reply:
> Are we accidentally emitting duplicate and inconsistent CreateTableEvents to the downstream

Yes. Updated and fixed it.